ic-agent/src/agent/http_transport/reqwest_transport.rs

```diff
@@ -236,7 +237,19 @@ impl ReqwestTransport {
             }
         }
 
-        if status.is_client_error() || status.is_server_error() {
+        // status == OK means we have an error message for call requests
+        // see https://internetcomputer.org/docs/current/references/ic-interface-spec#http-call
+        if status == StatusCode::OK && endpoint.ends_with("call") {
```

Comment:
It looks like a replica error (with reject_code, reject_message, and error_code) also come from an endpoint that ends in `/query`: https://internetcomputer.org/docs/current/references/ic-interface-spec#http-query

Reply:
Yes, that is true. However, it's only for `/call` where an HTTP status code of 200 means that the response body is a replica error. The specification for `/query` has not changed in the spec change and the current approach for detecting replica errors is valid. There is an existing integration test covering the case where we have replica errors for query calls.